### PR TITLE
cmd: crio: main: remove socket multiplexing

### PR DESCRIPTION
**Pull request by runcom**
`cmux` is leaking goroutines when using OpenShift and the cadvisor integration, probably because of the really high load. I've opened a PR upstream for this https://github.com/soheilhy/cmux/issues/55 while still trying to figure this out ourselves.

Because of the above, it makes sense to remove the multiplexer on the socket and have the info endpoint listens on `/var/run/crio/info.sock`. This PR just does that.

For the cadvisor integration, we need the following patch instead:
```diff
diff --git a/container/crio/client.go b/container/crio/client.go
index b588552..9dcafaa 100644
--- a/container/crio/client.go
+++ b/container/crio/client.go
@@ -24,7 +24,7 @@ import (
 )

 const (
-       CrioSocket            = "/var/run/crio.sock"
+       CrioSocket            = "/var/run/crio/info.sock"
        maxUnixSocketPathSize = len(syscall.RawSockaddrUnix{}.Path)
 )
```

This patch should alleviate (a lot) the spikes and memory usage increase we saw in performance testing.

@derekwaynecarr @mrunalp PTAL

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

File: cmd/crio/main.go [hunk 1]
```diff
@@ -28,6 +27,10 @@ import (
 // It will be populated by the Makefile.
 var gitCommit = ""
 
+const (
+	crioInfoSocket = "/var/run/crio_info.sock"
```
Comment (rhatdan) [inline cmd/crio/main.go]: Lets move this to /var/run/crio/info.sock
We want to move the crio socket into that directory also,. so lets not make it harder on ourselves

**Comment by rhatdan** [comment]
Lets make it
/var/run/crio/info.sock

**Comment by runcom** [comment]
Alright, makes sense, I'll change it later today Dan!

**Comment by runcom** [comment]
I also need to update tests for this.

**Comment by mrunalp** [comment]
:+1: I think we should do this to get rid of unnecessary overhead of cmux inspecting packets.

File: cmd/crio/main.go (hunk 1 above)
Comment (runcom) [reply cmd/crio/main.go]: fixed Dan

**Comment by runcom** [comment]
Updated

**Comment by runcom** [comment]
No more goroutines leak with this patch:
```
# TYPE go_goroutines gauge
go_goroutines 28
```

@mrunalp @rhatdan PTAL

**Comment by TomSweeneyRedHat** [comment]
drats, something's up with the travis integration.  Looks like some kind of socket mishap.

**Comment by runcom** [comment]
Tests should be green now

**Comment by mrunalp** [comment]
Nice! We also need a PR for 1.0.2

**Comment by runcom** [comment]
> Nice! We also need a PR for 1.0.2

let's see if we can fix cmux for 1.0.2 - going to try a potential fix for 1.0.2 that doesn't require us to change the socket path.

**Comment by mrunalp** [comment]
Please don't merge as we are looking at the #1090 / #1091 instead.

**Comment by rhatdan** [comment]
We might want to do some splitting out of the API into different sockets so that we could allow processes access to the monitoring socket versus the primary socket to start/stop pods.

**Comment by runcom** [comment]
Yeah, though performances are back to normal now, we might want to revisit this later, maybe for 1.8 or better 1.9

Closing for now.

**Comment by rhatdan** [comment]
Yes,  Perhaps we could differentiate watching events and logs from the ability to manage PODS/Containers.
Even having a read/only interface versus a Read/Write.

**Pull request closed without merging**